### PR TITLE
[BUG, MRG] Fix biosemi example coordinate frames

### LINKED ISSUE #9641: Regression in "How to plot topomaps the way EEGLAB does" example - montage.get_positions() returns different positions than the channel positons in evoked object.
Hi, I just noticed that the [How to plot topomaps the way EEGLAB does](https://mne.tools/dev/auto_examples/visualization/eeglab_head_sphere.html) example no longer shows correct EEGLAB-like channel layout:

the two layouts are the same on the current development and stable versions (0.23), which is incorrect. On version 0.22 these layouts are different (and correct):

The sphere parameters used in the example are not the same as before and this seems due to a change in how channel positions are obtained. Previously they were dug out from `.info['chs']` dictionary, now `montage.get_positions()` is used. However these two are not exchangable because the applied montage is in head coordinates, while the positions returned by `.get_positions()` show 'unknown' coord_frame.
In the same example doing:
```python
pos_mntg = biosemi_montage.get_positions()['ch_pos']['Oz']
ch_idx = fake_evoked.ch_names.index('Oz')
pos_evkd = fake_evoked.info['chs'][ch_idx]['loc'][:3]

print([f'{v:0.5f}' for v in pos_mntg])
print([f'{v:0.5f}' for v in pos_evkd])
```
gives different positions:
```
['0.00000', '-0.09494', '-0.00332']
['0.00000', '-0.09494', '0.03683']
```
I'm not sure what the best solution should be - is the intent of `montage.get_positions()` is to give channel positions in unknown coord frame? If so, the example should be reverted to use the old and ugly `.info['chs'][idx]['loc'][:3]` way of obtaining channel positions.

### PR DESCRIPTION
**Pull request by alexrockhill**
Fixes #9641.

### CODE REVIEW

**Review by mmagnuski — APPROVED** [review]
Great, thanks for a quick reaction. I'm a little behind with this `get_montage()` functionality :)

**Comment by alexrockhill** [comment]
I just cleaned it up a bit

File: examples/visualization/eeglab_head_sphere.py [hunk 1]
```diff
@@ -62,7 +62,10 @@
 
 # first we obtain the 3d positions of selected channels
 chs = ['Oz', 'Fpz', 'T7', 'T8']
-pos = np.stack([biosemi_montage.get_positions()['ch_pos'][ch] for ch in chs])
+# we need the montage transformed to "head" coordinates
```
Comment (mmagnuski) [inline examples/visualization/eeglab_head_sphere.py:65]: If we are adding more explanation, then maybe something like this:
```suggestion
# we want to have the channel positions in "head" coordinates - we can get
# these from the ``Evoked`` object via ``.get_montage()``
```

**Comment by mmagnuski** [comment]
Great! I'll go ahead and merge.

**Pull request merged**